src/check_update.ts

```diff
@@ -38,9 +38,14 @@ export interface Release {
   } | null;
 }
 
+interface InaccessibleFile {
+  name: string;
+  error: Error;
+}
+
 export const checkForSDKUpdates = async () => {
-  const versionMap: VersionMap = await createVersionMap();
-  const updateResp = createUpdateResp(versionMap);
+  const { versionMap, inaccessibleFiles } = await createVersionMap();
```

Comment:
Let's start at the very top: in order for us to bubble up the information from the individual files that were inaccessible in an attempt to both retrieve and update, a pretty deep drill down and passing around of that information was required.